### PR TITLE
Custominvoice amount, fixes #232 

### LINKED ISSUE #232: Custominvoice, allow user to define how much they want to pay
At the moment custominvoice defaults to the recommended price. Allow user to define how much they want to pay to minimize confusion. Of course the price must be checked to be valid.

### PR DESCRIPTION
**Pull request by tswfi**
allow user to define the price to be paid per unit during creation of custom invoice

### CODE REVIEW

**Review by olmari — CHANGES_REQUESTED** [review]
Everything else works as should exactly to set limits, but for services without minimum defined it will accept 0 and negative numbers, which is "illegal" in the context.

File: www/forms.py [hunk 1]
```diff
@@ -82,16 +82,56 @@ class FileImportForm(forms.Form):
...
+        And check that the price matches with the selected service
+        """
+        cleaned_data = super().clean()
+        subscription = cleaned_data["service"]
+        service = subscription.service
+        price = cleaned_data["price"]
+        if(service.cost_max and price > service.cost_max):
+            raise forms.ValidationError(
+                _('Price cannot be above service max cost: %(max)s'),
+                params={'max': service.cost_max},
```
Comment (olmari) [inline www/forms.py:122]: Needs adding some sanity checking against some minimum real money value, obviously can't be zero or negative, so "over zero" would propably be exactly correct.

**Comment by tswfi** [comment]
ce46995 adds a check for minimum of 0.01 for the price.

**Review by olmari — APPROVED** [review]
Everything works, including zero sum check, tested all possible values around boundaries to the cent.

**Pull request merged**